src/io/flutter/logging/FlutterLogTree.java

```diff
@@ -506,30 +521,48 @@ public TableCellRenderer getCellRenderer(int row, int column) {
     return model.columns.getRenderer(column);
   }
 
-  public void setFilter(@Nullable EntryFilter filter) {
-    // Only set and reload if the filter has changed.
-    if (!Objects.equals(this.filter, filter)) {
-      this.filter = filter;
-      reload();
+  public void setFilter(@NotNull FlutterLogFilterPanel.FilterParam filter) {
+    if (Objects.equals(this.filter, filter)) {
+      return;
     }
+    this.filter = filter;
+    updateRowFilter(filter);
+    updateCounter();
   }
 
-  void reload() {
-    ApplicationManager.getApplication().invokeLater(() -> {
-      model.getRoot().removeAllChildren();
-
-      final List<FlutterLogEntry> entries = model.log.getEntries();
-      final List<FlutterLogEntry> matched = entries.stream()
-        .filter(entry -> filter == null || filter.accept(entry)).collect(Collectors.toList());
+  private void updateRowFilter(@NotNull FlutterLogFilterPanel.FilterParam filter) {
+    final String nonNullExpression = filter.getExpression() == null ? "" : filter.getExpression();
+    final String quoteRegex = filter.isRegex() ? nonNullExpression : Pattern.quote(nonNullExpression);
+
+    final String matchCaseRegex = filter.isMatchCase() ? "" : "(?i)";
+    final String standardRegex = matchCaseRegex + "(?s).*" + quoteRegex + ".*";
+    final RowFilter<TableModel, Object> logMessageFilter;
```

Comment:
We all use regex here.
+ Match case => option `(?i)`.
+ Don't use regex => `Pattern.quote` 
+ Use regex => Keep original user input.